### PR TITLE
feat: librarian

### PR DESCRIPTION
**Pull request by a-frantz**
The `librarian` tool is still under development (see https://github.com/DesmondWillowbrook/Librarian/issues/17), but there are passenger-changes I added to this branch I want to see in `main`. So I'm voting for a preemptive merge, and we can update `librarian` later.

The passenger change in question: changing how analyses are toggled on/off in QC. By this, I mean we now present a "base set of analyses" with a number of "extra analyses" that can be **toggled on**, but are **off by default**. This is more-or-less how it currently works, though we had a variety of "toggling methods" and it was awkward IMO.

In practice, the above means dropping the `"DNA"/"RNA"` `String molecule` parameter and replacing it with `Boolean rna = false`. And some tweaking of phrasing in the docs.

### CODE REVIEW

File: tools/librarian.wdl [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Float read1_size = size(read_one_fastq, "GiB")
+    Int disk_size_gb = (
+        ceil(read1_size) + 10 + modify_disk_size_gb
+    )
+
+    command <<<
+        set -euo pipefail
+
+        mkdir ~{prefix}
+        RUST_LOG=trace /app/librarian --local --raw -o ~{prefix} ~{read_one_fastq}
```
Comment (adthrasher) [inline tools/librarian.wdl:39]: Do we need `RUST_LOG=trace` in production?

File: tools/librarian.wdl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    >>>
+
+    output {
+        File report = "~{prefix}.tar.gz"
+        File raw_data = "~{prefix}/librarian_heatmap.txt"
+    }
+
+    runtime {
+        memory: "4 GB"
+        disk: "~{disk_size_gb} GB"
```
Comment (adthrasher) [inline tools/librarian.wdl:51]:
```suggestion
        disks: "~{disk_size_gb} GB"
```

File: workflows/qc/quality-check-standard.wdl [hunk 3]
```diff
@@ -402,21 +409,21 @@ task parse_input {
...
+            check: "Dummy output to indicate success and to enable call-caching",  # TODO: is this still needed with labels?
             labels: "An array of labels to use on the result coverage files associated with each coverage BED"
         }
     }
 
     parameter_meta {
         coverage_labels: "An array of equal length to `coverage_beds_len` which determines the prefix label applied to coverage output files. If an empty array is supplied, defaults of `regions1`, `regions2`, etc. will be used."
-        input_molecule: "Must be `DNA` or `RNA`"
-        gtf_provided: "Was a GTF supplied by the user? Must be `true` if `input_molecule = RNA`."
+        rna: "Is the input molecule RNA?"
```
Comment (adthrasher) [inline workflows/qc/quality-check-standard.wdl:419]: What about `input` -> `sequenced`? I think that's more accurate. It is input to this workflow though, but the molecule is what was sequenced.

**Review by adthrasher — CHANGES_REQUESTED** [review]

File: tools/librarian.wdl (hunk 1 above)
Comment (a-frantz) [reply tools/librarian.wdl:39]: Good catch.
(For the record, the librarian team has ignored my suggestion to put the binary in the image's PATH. Which is why it's accessed `/app/librarian`)

**Pull request merged**